src/accelerate/test_utils/training.py

```diff
@@ -21,11 +21,10 @@
 
 class RegressionDataset:
     def __init__(self, a=2, b=3, length=64, seed=None):
-        if seed is not None:
-            np.random.seed(seed)
+        rng = np.random.default_rng(seed)
         self.length = length
-        self.x = np.random.normal(size=(length,)).astype(np.float32)
-        self.y = a * self.x + b + np.random.normal(scale=0.1, size=(length,)).astype(np.float32)
+        self.x = rng.normal(size=(length,)).astype(np.float32)
+        self.y = a * self.x + b + rng.normal(scale=0.1, size=(length,)).astype(np.float32)
```

Comment:
Alternative suggestion here, what if we modified `set_seed` to give it a `default_rng` bool flag, and when `True` will call `np.random.default_rng` instead and return the object.

Then, `RegressionDataset` checks if `seed` is an `int` or a generator, and then modifies accordingly towards what we want to do. 

E.g. something like:

```python
rng = np.random
if seed is not None:
    if isinstance(seed, int):
        np.random.seed(seed)
    else:
        rng = seed
```

wdyt @sgugger? This way we maintain the original behavior, and allow for this when needed (aka, only TPUs)

Reply:
That doesn't quite work as `set_seed` sets multiple seeds. Why would it return a Numpy RNG and not a torch rng?